docs/static/persistent-queues.asciidoc

```diff
@@ -1,44 +1,229 @@
+`queue.drain`:: Specify `true` if you want Logstash to wait until the persistent queue is drained before shutting down. The amount of time it takes to drain the queue depends on the number of events that have accumulated in the queue. Therefore, you should avoid using this setting unless the queue, even when full, is relatively small and can be drained quickly. 
+`queue.max_events`:: The maximum number of events not yet read by the pipeline worker. The default is 0 (unlimited).
+We use this setting for internal testing. 
+Users generally shouldn't be changing this value.
+`queue.max_bytes`:: The total capacity of _each queue_ in number of bytes. 
+Unless overridden in `pipelines.yml` or central management, each persistent
+queue will be sized at the value of `queue.max_bytes` specified in
+`logstash.yml`. 
+The default is 1024mb (1gb).
++
+Be sure that your disk has sufficient capacity to handle the cumulative total of `queue.max_bytes` across all persistent queues.
+The total of `queue.max.bytes` for _all_ queues should be
+lower than the capacity of your disk. 
++
+TIP: If you are using persistent queues to protect against data loss, but don't
+require much buffering, you can set `queue.max_bytes` to a smaller value, such
+as 10mb.
+A smaller value produces smaller queues and improves queue performance. 
+
+`queue.checkpoint.acks`:: Sets the number of acked events before forcing a checkpoint. 
+Default is `1024`. Set to `0` for unlimited.
+`queue.checkpoint.writes`:: Sets the maximum number of written events before a forced checkpoint. 
+Default is `1024`. Set to `0` for unlimited.
++
+To avoid losing data in the persistent queue, you can set `queue.checkpoint.writes: 1` to force a checkpoint after each event is
+written. Keep in mind that disk writes have a resource cost. Setting this value
+to `1` ensures maximum durability, but can severely impact performance.
+See <<durability-persistent-queues>> to better understand the trade-offs.
+`queue.checkpoint.interval`:: Sets the interval in milliseconds when a checkpoint is forced on the head page.
+Default is `1000`. Set to `0` to eliminate periodic checkpoints.
+
+[[pq-config-notes]]
+==== Configuration notes
+
+Every situation and environment is different, and the "ideal" configuration varies.
+If you optimize for performance, you may increase your risk of losing data.
+If you optimize for data protection, you may impact performance. 
+
+[[pq-size]]
+===== Queue size
+You can control queue size with the `queue.max_events` and  `queue.max_bytes` settings.
+If both settings are specified, Logstash uses whichever criteria is reached
+first. 
+See <<backpressure-persistent-queue>> for behavior when queue limits are
+reached.
+
+[[pq-lower-max_bytes]]
+===== Smaller queue size
+If you are using persistent queues to protect against data loss, but don't
+require much buffering, you can set `queue.max_bytes` to a smaller value, such
+as 10mb.
```

Comment:
I'm not sure I'd necessarily put a concrete value here, its another one of those "it depends" type of values.